cli/src/pcluster/resources/custom_resources/custom_resources_code/budget_triggered_stop.py

```diff
@@ -0,0 +1,78 @@
+import datetime
+import os
+import time
+
+import boto3
+from boto3.dynamodb.conditions import Attr
+
+DB_KEY = "COMPUTE_FLEET"
+DB_DATA = "Data"
+
+COMPUTE_FLEET_STATUS_ATTRIBUTE = "status"
+COMPUTE_FLEET_LAST_UPDATED_TIME_ATTRIBUTE = "lastStatusUpdatedTime"
+
+
+def update_item(table, status, current_status):
+    table.update_item(
+        Key={"Id": DB_KEY},
+        UpdateExpression="set #dt.#st=:s, #dt.#lut=:t",
+        ExpressionAttributeNames={
+            "#dt": DB_DATA,
+            "#st": COMPUTE_FLEET_STATUS_ATTRIBUTE,
+            "#lut": COMPUTE_FLEET_LAST_UPDATED_TIME_ATTRIBUTE,
+        },
+        ExpressionAttributeValues={
+            ":s": str(status),
+            ":t": str(datetime.datetime.now(tz=datetime.timezone.utc)),
+        },
+        ConditionExpression=Attr(f"{DB_DATA}.{COMPUTE_FLEET_STATUS_ATTRIBUTE}").eq(str(current_status)),
+    )
+
+
+def update_status_with_last_updated_time(table_name, region, status):
+    try:
+        table = boto3.resource("dynamodb", region_name=region).Table(table_name)
+        current_status = get_dynamo_db_data(table).get(COMPUTE_FLEET_STATUS_ATTRIBUTE)
+        print("Stopping Compute Fleet - Triggered by Budget Threshold.")
+        if current_status == status:
+            return
+        elif current_status == "RUNNING" or current_status == "PROTECTED":
```

Comment:
nit: current_status in ["RUNNING", "PROTECTED"]

Reply:
Updated!